webview/src/setup/components/SetupContainer.tsx

```diff
@@ -16,33 +19,32 @@ const getTooltipIconType = (isSetup: boolean, isConnected = true) => {
 
 export const SetupContainer: React.FC<{
   children: React.ReactNode
-  sectionCollapsed: typeof DEFAULT_SECTION_COLLAPSED
   sectionKey: SetupSection
-  setSectionCollapsed: (value: typeof DEFAULT_SECTION_COLLAPSED) => void
   title: string
   isSetup: boolean
   isConnected?: boolean
-}> = ({
-  children,
-  sectionCollapsed,
-  sectionKey,
-  setSectionCollapsed,
-  title,
-  isSetup,
-  isConnected
-}) => (
-  <SectionContainer
-    sectionCollapsed={sectionCollapsed[sectionKey]}
-    sectionKey={sectionKey}
-    title={title}
-    icon={getTooltipIconType(isSetup, isConnected)}
-    onToggleSection={() =>
-      setSectionCollapsed({
-        ...sectionCollapsed,
-        [sectionKey]: !sectionCollapsed[sectionKey]
-      })
-    }
-  >
-    {children}
-  </SectionContainer>
-)
+}> = ({ children, sectionKey, title, isSetup, isConnected }) => {
+  const sectionCollapsed =
+    useSelector((state: SetupState) => state.webview.sectionCollapsed) ||
+    DEFAULT_SECTION_COLLAPSED
+  const dispatch = useDispatch()
+
+  return (
+    <SectionContainer
+      sectionCollapsed={sectionCollapsed[sectionKey]}
+      sectionKey={sectionKey}
+      title={title}
+      icon={getTooltipIconType(isSetup, isConnected)}
+      onToggleSection={() =>
+        dispatch(
+          updateSectionCollapsed({
```

Comment:
There could be another action just to toggle a section. That would simplify things here and let the action handle the state.